src/clj/puppetlabs/services/config/puppet_server_config_core.clj

```diff
@@ -107,6 +134,12 @@
    (let [config (get-puppet-config* jruby-puppet)]
      (assoc config :puppet-version (.puppetVersion jruby-puppet)))))
 
+(defn resolve-ca-settings
+  [puppet-config server-ca-settings]
+  (when-let [cadir (:cadir server-ca-settings)]
+    (-> (interpolate-cadir cadir puppet-config)
+        (merge server-ca-settings))))
```

Comment:
Maybe we don't do the merge here and just return puppet's config hash with `cadir` interpolated? And then call this somewhere before we return the hash? It does seem a bit weird (like maybe hard to debug) to do that to something we're going to go around thinking of as settings out of puppet...